sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -500,12 +442,110 @@ object GpuDeviceManager extends Logging {
     }
     if (!PinnedMemoryPool.isInitialized && pinnedSize > 0) {
       logInfo(s"Initializing pinned memory pool (${pinnedSize / 1024 / 1024.0} MiB)")
-      PinnedMemoryPool.initialize(pinnedSize, gpuId, setCuioDefaultResource)
+      PinnedMemoryPool.initialize(pinnedSize, gpuId, conf.pinnedPoolCuioDefault)
     }
     // Host memory limits must be set after the pinned memory pool is initialized
     HostAlloc.initialize(nonPinnedLimit)
   }
 
+  // visible for testing
+  def getPinnedPoolAndOffHeapLimits(conf: RapidsConf, sparkConf: SparkConf,
+                                            memCheck: MemoryChecker =
+                                            MemoryCheckerImpl): (Long, Long) = {
+    val perTaskOverhead = conf.perTaskOverhead
+    val totalOverhead = perTaskOverhead * GpuDeviceManager.numCores
+    val confPinnedSize = conf.pinnedPoolSize
+    val confLimit = conf.offHeapLimit
```

Comment:
this PR stops supporting `offHeapLimitEnabled`? Are we deprecating that or should we honor it in some way?

Reply:
Meaning I think we want a way to turn off the limits and go back to status quo, at least in the interim. Thoughts?